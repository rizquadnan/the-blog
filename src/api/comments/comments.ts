export type TComment = {
  id: number;
  post_id: number;
  name: string;
  email: string;
  body: string;
};
