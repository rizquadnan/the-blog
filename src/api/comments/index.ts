export * from "./comments";
export * from "./useComments";
