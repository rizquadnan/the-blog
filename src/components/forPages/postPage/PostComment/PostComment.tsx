import { VStack } from '@/components/uiKit/VStack'
import { SkeletonText, Text } from '@chakra-ui/react'
import React from 'react'

type TPostComment = {
  author: {
    name: string
    email: string
  }
  content: string
  isLoading?: boolean
}
export function PostComment(props: TPostComment) {
  return (
    <VStack spacing="12px" minW="300px">
      <VStack spacing="4px">
        {props.isLoading ? (
          <SkeletonText noOfLines={1} fontSize="18px" maxW="100px" />
        ) : (
          <Text fontSize="16px" fontWeight={600}>
            {props.author.name}
          </Text>
        )}
        {props.isLoading ? (
          <SkeletonText noOfLines={1} fontSize="12px" maxW="120px" />
        ) : (
          <Text fontSize="12px" opacity={0.7}>
            {props.author.email}
          </Text>
        )}
      </VStack>

      {props.isLoading ? (
        <SkeletonText noOfLines={1} />
      ) : (
        <Text fontSize="14px">{props.content}</Text>
      )}
    </VStack>
  )
}

export default PostComment
