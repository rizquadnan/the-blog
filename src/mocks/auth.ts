export const MOCK_USER = {
  id: 271030,
  name: "Shashi Butt I",
  email: "shashi_butt_i@lowe.com",
  gender: "female",
  status: "inactive",
};
