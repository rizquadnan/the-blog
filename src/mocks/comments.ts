export const MOCK_COMMENTS = [
  {
    id: 13431,
    post_id: 17319,
    name: "Bhima Iyengar",
    email: "iyengar_bhima@ernser.info",
    body: "Quod dolor ea. Aut voluptas qui. Consequatur et laboriosam. Voluptatem nostrum occaecati.",
  },
  {
    id: 13423,
    post_id: 17305,
    name: "Saroja Sinha",
    email: "saroja_sinha@hackett.name",
    body: "Autem laudantium molestiae. Laborum dignissimos optio. Optio et dicta. Earum aut quia.",
  },
  {
    id: 13419,
    post_id: 17299,
    name: "Meena Sinha",
    email: "sinha_meena@crona.name",
    body: "Qui cupiditate necessitatibus. Consequatur sed aspernatur.",
  },
  {
    id: 13381,
    post_id: 17259,
    name: "Preity Tagore",
    email: "tagore_preity@reynolds.info",
    body: "Quo aut nihil. Eveniet sit corporis. Autem ducimus et.",
  },
  {
    id: 13379,
    post_id: 17255,
    name: "Yogesh Dwivedi MD",
    email: "dwivedi_yogesh_md@flatley.net",
    body: "Facilis natus pariatur. Optio maxime dicta. Dolor temporibus soluta. Ut dicta quisquam.",
  },
  {
    id: 13378,
    post_id: 17254,
    name: "Dayaamay Talwar",
    email: "talwar_dayaamay@hansen.name",
    body: "Ut aut cupiditate.",
  },
  {
    id: 13376,
    post_id: 17251,
    name: "Chandani Gupta Jr.",
    email: "gupta_chandani_jr@stiedemann.com",
    body: "Aut ut atque. Omnis esse sequi. Id porro consectetur. Animi veritatis aperiam.",
  },
  {
    id: 13375,
    post_id: 17249,
    name: "Ajit Patil",
    email: "ajit_patil@white-ratke.info",
    body: "Ipsa ea quia. Voluptas voluptatem eaque. Magni vel eligendi.",
  },
  {
    id: 13374,
    post_id: 17248,
    name: "Vasudev Kapoor",
    email: "kapoor_vasudev@roob.org",
    body: "Explicabo voluptatem est. Consequatur sed quia.",
  },
  {
    id: 13371,
    post_id: 17244,
    name: "Omana Nambeesan MD",
    email: "omana_nambeesan_md@grady.co",
    body: "Sapiente mollitia consequatur. Dolores aut provident. Quia cupiditate rerum. Ut eos quos.",
  },
];
