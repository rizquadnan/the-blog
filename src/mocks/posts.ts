export const MOCK_POSTS = [
  {
    id: 17218,
    user_id: 269828,
    title:
      "Tergeo acquiro contigo acceptus officiis pel centum assentator aiunt eos civis statim venio umbra.",
    body: "Accusator tutis eos. Utrimque beatae tres. Civis ante voluptatem. Curis adimpleo occaecati. Nesciunt copia suasoria. Spiritus carpo trucido. Truculenter cui tutis. Vel unus altus. Pel teneo supplanto. Eum patior natus. Cariosus thymum solum. Curvo comitatus coepi. Subito capillus sumo. Benigne ver pecto. Uxor conor agnitio. Natus cruciamentum nam. Statua ut cerno. Videlicet caries curvo.",
  },
  {
    id: 17216,
    user_id: 269824,
    title: "Abutor addo color nostrum talus alveus claudeo ad neque et cubo.",
    body: "Terra admitto auxilium. Angulus deputo provident. Adstringo sum combibo. Certo depromo qui. Tamquam paens crustulum. Temperantia in carpo. Attonbitus est spiculum. Vorax cogo denuncio. Et et vultuosus. Quis tricesimus et. Theologus claudeo aufero. Aut altus viscus. Paulatim triginta tribuo. Aeternus adulescens vox. Curatio trans volo. Patior surgo consequuntur. Canto sto creator. Universe illo admoveo. Quod tamisium demoror. Cubicularis conatus comitatus. Tabesco officiis sum. Voro aer quo.",
  },
  {
    id: 17215,
    user_id: 269821,
    title:
      "Velut ea virgo vicinus velit super coniecto valeo censura degusto clamo eaque ea et vinco voluptas.",
    body: "Arca advoco quasi. Turba ullus amicitia. Coruscus varietas spero. Absorbeo alii eveniet. Decretum consequatur suasoria. Tremo unde auctor. Antea dolor conservo. Adipiscor arceo voluptas. Autem occaecati sit. Adnuo voluptatem cribro. Dolore laborum velit. Crebro capio delinquo. Aliquam tabgo cohibeo. Coma sophismata pecus. Temeritas consequatur aestivus. Viduata traho teres. Curvus vel creator. Bardus voluptates arguo.",
  },
  {
    id: 17214,
    user_id: 269819,
    title: "Aeneus clamo alveus ea compono civis temeritas conatus.",
    body: "Comes suffoco auris. Verus acsi socius. Et vigilo altus. Qui eum celo. Casso usque arbustum. Curso temptatio praesentium. Beatus torqueo deficio. Vis acceptus cenaculum. Absconditus aggero caries. Carus villa bos. Charisma utrimque textilis. Nam volutabrum comes. Trepide canis stillicidium. Curso vespillo sub. Comes tantillus cernuus. Callide quam uterque. Deporto adduco trepide. Ventus synagoga admoneo. Trans angelus natus. Carmen soluta harum.",
  },
  {
    id: 17213,
    user_id: 269818,
    title:
      "Et ea venustas teres audio concedo uterque avaritia architecto coadunatio villa suus asper creptio avarus quia bos.",
    body: "Vir aut alias. Arcus ocer corrumpo. Exercitationem officia dedecor. Cohibeo veritas absorbeo. Amoveo tantum tepesco. Tandem vinitor enim. Viscus consequatur cribro. Undique uxor talio. Sordeo cicuta tabernus. Decerno carpo ascisco. Caries atavus utilis. Sollicito somniculosus deorsum. Vilicus carpo virgo. Administratio sunt textus. Alias minus peior. Inventore nulla administratio. Tonsor carcer admiratio. Iusto sponte caute. Civitas ut vinco. Et adhaero aut. Cui nihil valetudo.",
  },
  {
    id: 17212,
    user_id: 269814,
    title:
      "Capto curo congregatio strues tergum despirmatio thesis barba defessus claustrum consequatur ascisco vito decimus advenio desino totam.",
    body: "Dolorem crinis quam. Defero quia uredo. Addo via laborum. Tamquam acies decet. Comparo non subito. Viscus trucido animus. Circumvenio curvus vigilo. Tego deludo ut. Adhuc dolorem doloremque. Concido adimpleo decumbo. Comis ago colligo. Adhaero audentia totidem. Illum cognomen viridis. Deludo tergum convoco. Ut viduata speculum. Vox deleo caelestis. Chirographum aut defessus. Desolo substantia sopor. Iusto deleniti sequi. Sponte consequatur amet.",
  },
  {
    id: 17211,
    user_id: 269811,
    title: "Tui tristis hic tepesco aduro aeneus.",
    body: "Claudeo causa tutis. Ut textilis provident. Aperte conservo tepesco. Demoror cum tempore. Quo vitiosus expedita. Ascisco degenero advoco. Calculus talus vitium. Ascisco acquiro textilis. Vociferor celer coerceo. Cui apto adopto. Constans volaticus patria. Consuasor comprehendo aranea. Aiunt vociferor vicissitudo. Studio reiciendis non. Illum supra deprimo. Tabesco pecunia suffoco. Benevolentia tam vulnus. Tempus rerum cura.",
  },
  {
    id: 17210,
    user_id: 269811,
    title: "Eligendi auctus canonicus viduata adicio valeo.",
    body: "Calculus deduco vitium. Non vilicus conforto. Adflicto colo adsum. Aeger tego centum. Vilicus beatus tamisium. Agnosco corroboro adulescens. Calculus deserunt spero. Deleniti cruentus blanditiis. Utique quo vallum. Terror infit umbra. Depromo tener deserunt. Admiratio vita succurro. Crustulum celebrer bos. Cultura dens virtus. Coadunatio aveho textilis. Et cupio quod. Nemo suppellex tibi. Denuo distinctio aro. Cometes omnis cribro. Ad vis quia. Cogo suppellex sapiente. Suasoria cotidie quod.",
  },
  {
    id: 17209,
    user_id: 269810,
    title: "Depereo derideo caput canonicus stips dolorem quam tum.",
    body: "Sit ago succurro. Autem despirmatio administratio. Illum calcar sumptus. Crapula aeternus ipsa. Talus vomer abundans. Civitas ante conscendo. Turba vitium consequatur. Defungo centum summa. Consuasor derelinquo video. Tener addo decerno. Delinquo ulciscor optio. Surculus talio vehemens. Benevolentia demitto subiungo. Soleo et color. Congregatio denique defessus. Atavus non vaco. Eius sodalitas appello. Vigor paens minus.",
  },
  {
    id: 17208,
    user_id: 269809,
    title:
      "Vacuus versus sit usus crinis aliquid talus quae natus decimus vulpes decerno.",
    body: "Reiciendis articulus et. Sunt aptus agnosco. Copiose pax itaque. Suggero cernuus qui. Caritas alienus causa. Vomica sursum et. Claro communis arbustum. Et tibi alter. Turpe ut assumenda. Unus conatus tactus. Assentator utor creptio. Demoror brevis ut. Viduata comprehendo termes. Textor sol cursim. Quisquam triumphus voluntarius. Distinctio sunt adficio. Adhaero inflammatio amaritudo. Stella crur adsuesco. Vita aetas delinquo. Omnis cum angustus. Totam brevis adnuo. Tunc vilitas congregatio.",
  },
  {
    id: 17207,
    user_id: 269805,
    title:
      "Careo voluptas cubicularis comitatus cernuus cunctatio articulus abstergo.",
    body: "Aestus clam tempus. Veritas termes pauper. Aqua decor decumbo. Virtus arma volva. Ipsa derideo talio. Asper rerum aeternus. Quia vetus curtus. Sed stella cornu. Culpa uredo supellex. Basium audeo tardus. Comminor concido delinquo. Debitis acervus celo. Video sub alo. Cedo commodi subnecto. Aut non animadverto. Truculenter bibo accusamus. Astrum credo utpote. Abduco angustus video. Adimpleo cribro carpo.",
  },
  {
    id: 17206,
    user_id: 269805,
    title:
      "Acies adflicto tabernus communis vulgus adopto adaugeo arbitro expedita.",
    body: "Suadeo cras sortitus. Atavus defungo crastinus. Terminatio viduo altus. Quo theologus ter. Auditor et adipisci. Vorax venustas adultus. Neque ullus conspergo. Thermae decimus crapula. Possimus aestivus arbor. Iusto error quo. Vivo crudelis nesciunt. Velum infit usus. Dolores absorbeo certo. Quo volubilis celer. Urbanus absum solus. Termes voluptatum adsuesco. Repudiandae vos defessus. Tener vespillo vita. Crastinus delectus quis. Vehemens traho civis.",
  },
  {
    id: 17205,
    user_id: 269803,
    title:
      "Ago abduco cognatus advoco sint surculus attero fuga ciminatio cui velociter angelus.",
    body: "Amicitia asporto pecunia. Triduana video quis. Audentia vulgus vel. Possimus angustus cresco. Valetudo vitium aedificium. Paulatim avoco utrum. Debeo vix ater. Sponte angustus vel. Curso urbanus trans. Cetera dolor at. Civitas volo aeternus. Et pecto vesper. Sol tabula consequatur. Ascit cornu advoco. Certe degero vetus. Alias cribro quaerat. Similique eum strues. Deinde amplexus cribro.",
  },
  {
    id: 17204,
    user_id: 269801,
    title:
      "Volaticus corpus vir virgo trans ultio eveniet nobis amo aedificium dolore autem.",
    body: "Vicinus aqua virga. Ter acer celer. Villa et auxilium. Tredecim allatus sollers. Quaerat aut capitulus. Attollo mollitia adeptio. Callide condico attonbitus. Vilicus clam cavus. Dolore assentator et. Stabilis celer vesco. Desidero tersus convoco. Aut sophismata peccatus. Qui ver ascisco. Avarus animadverto ut. Defaeco pauci arcesso. Cum officia ceno. Vere bellicus tabula. Terra est vicinus. Vallum tumultus aeternus.",
  },
  {
    id: 17203,
    user_id: 269800,
    title:
      "Uberrime thermae cibus creo qui utrimque abduco coepi adhuc tutis patior bellum iusto verbum colloco.",
    body: "Arceo reprehenderit adiuvo. Facere statim tolero. Et adfectus sub. Vinitor curia centum. Attero nam cum. Talis ademptio theatrum. Adfero tristis stella. Dedecor creptio auctor. Unus tredecim deludo. Carus aut taceo. Voluptatibus vereor velum. Verto quia delectatio. Earum acies desparatus. Deripio culpa mollitia. Civitas in cogo. Suscipio quia cur. Copia adfero demens. Verbera avoco cupiditate. Aut varius et. Alveus suppellex torqueo.",
  },
  {
    id: 17202,
    user_id: 269800,
    title:
      "Tandem voro cupressus consequatur spero angelus tamdiu ago virgo video comptus triduana inflammatio tardus sursum carmen velut accusamus appello.",
    body: "Sortitus tredecim cetera. Desidero minus contra. Arcus adinventitias totidem. Tam venio iusto. Viscus ultra dolorem. Amor officia infit. Tantillus varietas acquiro. Saepe thesaurus conqueror. Culpa conforto facere. Adiuvo terminatio talis. Creta perspiciatis bellum. Traho bardus averto. Conicio vorago amicitia. Angulus theologus custodia. Avaritia tantum spiculum. Ademptio iusto suspendo. Aegrotatio utpote aut. Eos avoco vulgaris. Vomica nostrum debitis. Tergum conservo tenuis. Vis tutis tunc.",
  },
  {
    id: 17201,
    user_id: 269792,
    title: "Caecus aufero contego deduco et vindico accusamus adultus.",
    body: "Labore qui nobis. Bene omnis soluta. Aut harum apud. Tepesco non temporibus. Cubitum cupiditas vesper. Conicio vereor certo. Viriliter contabesco in. Vulnero bibo amissio. Allatus suffragium vicinus. Ciminatio aduro aetas. Terreo tergum appositus. Angelus crudelis clarus. Cruentus tabgo cognatus. Spectaculum est tamquam. Audio arguo pauci. Tener deripio crapula. Conduco bellicus cervus.",
  },
  {
    id: 17200,
    user_id: 269792,
    title:
      "Dens patrocinor amoveo vulariter conspergo synagoga venio aegrotatio delectatio illum videlicet nihil admiratio depono patior carcer depromo via.",
    body: "Ventus tendo necessitatibus. Impedit verus vae. Qui atrocitas correptius. Vergo aureus ut. Adnuo antea crustulum. Vilicus bardus bibo. Vos vilitas adulescens. Animus thema cuppedia. Venia sufficio agnitio. Vinitor vulariter vis. Decimus virga caries. Delicate est spes. Aequitas tui callide. Compono spiritus chirographum. Contra aestivus fuga. Absorbeo suffoco capto.",
  },
  {
    id: 17117,
    user_id: 254897,
    title:
      "Calamitas cur certe ait bibo tenuis sol vinum vere arbor corrupti neque peccatus nihil.",
    body: "Cubo comptus super. Decumbo video mollitia. Arbor cunabula careo. Harum ait delego. Absens triduana nesciunt. Solium atavus valeo. Reprehenderit decens vulticulus. Vereor conatus turba. Articulus stips tutamen. Umquam tristis conatus. Statua damnatio aut. Patior aut aureus. Decet validus theologus. Bestia incidunt canis. Cohors cetera sint. Texo valeo statim. Cui sustineo accusator. Tero deorsum adhaero. Pariatur verto spoliatio. Demo suffragium quis.",
  },
  {
    id: 17095,
    user_id: 251787,
    title:
      "Triginta spectaculum succurro cibus praesentium celer sodalitas varietas ademptio crux.",
    body: "Deludo ancilla depromo. Consequatur auditor conservo. Unde quis deporto. Valeo vomito acerbitas. Caveo volup tribuo. Defessus conqueror apparatus. Tempora curtus statim. Vindico sum adicio. Aut crepusculum eos. Id sollicito crebro. Viriliter ventito synagoga. Tres cicuta sum. Tenus stipes alius. Temeritas copia demo. Contigo delego teres. Cubo minima accipio.",
  },
];
