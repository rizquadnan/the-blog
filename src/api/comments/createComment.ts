import { fetcher } from "../fetcher";
import { TCreateCommentPayload } from "./comments";

type TCreatePostArgs = {
  payload: TCreateCommentPayload;
  postId: number;
};
export function createComment(args: TCreatePostArgs) {
  return fetcher.post(`/posts/${args.postId}/comments`, args.payload);
}
