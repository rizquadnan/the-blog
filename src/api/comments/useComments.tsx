import { AxiosResponse } from 'axios'
import useSwrInfinite from 'swr/infinite'
import { fetcher, TPagination } from '../fetcher'
import {
  getLoadedUntilItemNumber,
  getTotalItemsFromResponse,
} from '../fetcher/utils'
import { TComment } from './comments'

type TUseCommentsReturnVal = {
  comments: TComment[]
  error: any
  isLoading: boolean
  refreshComments: () => void
  pagination: TPagination
}
type TUsePostsArgs = {
  postId: number | null
  perPage?: number
  onErrorCallback: (err: unknown) => void
}
export function useComments({
  postId,
  perPage = 5,
  onErrorCallback,
}: TUsePostsArgs): TUseCommentsReturnVal {
  const { data: res, error, isLoading, setSize, size, mutate } = useSwrInfinite<
    AxiosResponse<TComment[]>
  >(
    (index) =>
      postId
        ? `/posts/${postId}/comments?page=${index + 1}&per_page=${perPage}`
        : null,
    (url) => fetcher.get(url),
    {
      onError: onErrorCallback,
    },
  )

  const comments: TComment[] = []
  res?.forEach(({ data }) => {
    comments?.push(...data)
  })

  console.log(
    'getLoadedUntilItemNumber(size, perPage)',
    getLoadedUntilItemNumber(size, perPage),
  )
  console.log(
    'getTotalItemsFromResponse(res?.[0])',
    getTotalItemsFromResponse(res?.[0]),
  )

  return {
    comments,
    error,
    isLoading: Boolean(
      isLoading || (size > 0 && res && typeof res[size - 1] === 'undefined'),
    ),
    refreshComments: () => mutate(),
    pagination: {
      hasMore:
        getLoadedUntilItemNumber(size, perPage) <=
          getTotalItemsFromResponse(res?.[0]) &&
        comments.length !== getTotalItemsFromResponse(res?.[0]),
      totalItems: getTotalItemsFromResponse(res?.[0]),
      nextPage: () => setSize(size + 1),
      prevPage: () => setSize(size - 1),
    },
  }
}
