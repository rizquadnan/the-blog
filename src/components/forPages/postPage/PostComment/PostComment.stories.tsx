import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { PostComment } from './PostComment'
import { MOCK_COMMENTS } from '@/mocks/comments'

type TPostComment = React.ComponentProps<typeof PostComment>
export default {
  title: 'Data Display/PostComment',
  component: PostComment,
  parameters: {
    layout: 'centered',
  },
} as Meta<TPostComment>

const Template: Story<TPostComment> = (args) => <PostComment {...args} />

const requiredProps: TPostComment = {
  author: {
    name: MOCK_COMMENTS[0].name,
    email: MOCK_COMMENTS[0].email,
  },
  content: MOCK_COMMENTS[0].body,
}
export const Default = Template.bind({})
Default.args = requiredProps

export const Loading = Template.bind({})
Loading.args = {
  ...requiredProps,
  isLoading: true,
}
