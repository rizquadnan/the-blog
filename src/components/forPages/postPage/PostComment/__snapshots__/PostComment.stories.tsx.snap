// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Display/PostComment Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Bhima Iyengar",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "iyengar_bhima@ernser.info",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Quod dolor ea. Aut voluptas qui. Consequatur et laboriosam. Voluptatem nostrum occaecati.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "data-display-postcomment--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`Data Display/PostComment Loading smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": undefined,
  "name": "data-display-postcomment--loading",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
